### ISSUE TITLE
Firefox support

### ISSUE THREAD

**Issue by Irker** [Firehed/u2f-php#1]
@Firehed hi!

Now firefox implemented U2F support and I propose to tune your library code to support it.

Lets check what differ in Firefox:
1) Firefox sending errorCode with 0 value even if no errors occurs.  
[Diff](https://github.com/Irker/u2f-php/commit/10b8d3d9dd8225fae0fcbe3b6fde3a3413c1411a)
2) Firefox sending client data in different format: not all properties and in different order. But for verification we required same json string what we received from Firefox. So I removed original json string modification from code: 
[https://github.com/Irker/u2f-php/commit/402f6dd02a543f29d6900fe6cff4be6cd4fa3581](url) 

Sorry for not prepared pull-requests, but I'm not sure about how you want to fix second problem in your code, so just forked and fix it faster fo myself. (and without tests fixes).

Thank you!

**Comment by Firehed** [comment]
Hi @Irker,

Thanks for letting me know!

Doing a quick search, it looks like the Firefox implementation is only in the nightlies and might not be 100% stable - can you confirm if that's the case? It seems like there are currently some open tasks on [bugzilla](https://bugzilla.mozilla.org/show_bug.cgi?id=1065729). If it's not in the public stable release, I'd prefer to hold off on making any changes to the library at this time in case there are minor changes to their implementation.

It also looks like there are some updates to the protocol, which may explain some of the variance on Firefox. When I have some time I will try to look over the delta in more detail and see what it takes to update everything.

**Comment by Irker** [comment]
Yes, it's not 100% stable but already in stable version and enabling by flag in about:config .
But as you can see to support FF we required not very big changes, so I think we can implement FF support right now and then just correct code in future when FF will release it.

But I do not insist, so I didn't make pull-request immediately. Lets save this issue for future.

**Comment by Firehed** [comment]
Seems reasonable to me!

I think the best course of action is to leave the changes on a branch, and keep it tagged as a beta until Firefox support is no longer behind an about:config flag. At that time, the branch can land on master (making any final changes, if necessary) and get tagged as the next stable version.

How do you feel about that?

**Comment by Irker** [comment]
Yes, it will be nice. 
Lets do it. 
Please, can you create new branch and I will prepare pull-request.

**Comment by Firehed** [comment]
I've made the `firefox` branch - looking forward to your PR :)
